### PR TITLE
Add Migration section and improve text

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,11 +26,15 @@ the same idea to other signing algorithms<sup>1</sup>.
...
+  - [Implementation in the Ledger app](#implementation-in-the-ledger-app)
+  - [Implementation in CosmJS](#implementation-in-cosmjs)
   - [Migration](#migration)
+    - [Special case: Cosmos Hub](#special-case-cosmos-hub)
+    - [Special case: Custom SLIP44 coin type](#special-case-custom-slip44-coin-type)
   - [Other signing algorithms](#other-signing-algorithms)
   - [Notes](#notes)
 
-<!-- Added by: simon, at: Mi  2 Jun 2021 15:24:01 CEST -->
+<!-- Added by: simon, at: Di 15 Jun 2021 11:04:10 CEST -->
```
Comment (willclarktech) [inline README.md]: Interesting data leak.

File: README.md [hunk 2]
```diff
@@ -26,11 +26,15 @@ the same idea to other signing algorithms<sup>1</sup>.
...
+    - [Implementation in the Ledger app](#implementation-in-the-ledger-app)
+    - [Implementation in CosmJS](#implementation-in-cosmjs)
   - [Migration](#migration)
+    - [Special case: Cosmos Hub](#special-case-cosmos-hub)
+    - [Special case: Custom SLIP44 coin type](#special-case-custom-slip44-coin-type)
   - [Other signing algorithms](#other-signing-algorithms)
   - [Notes](#notes)
 
-<!-- Added by: simon, at: Mi  2 Jun 2021 15:24:01 CEST -->
+<!-- Added by: simon, at: Di 15 Jun 2021 11:50:21 CEST -->
```
Comment (willclarktech) [inline README.md:37]: Interesting data leak.

File: README.md [hunk 3]
```diff
@@ -309,7 +310,46 @@ export function makeSimpleHdPath(chainIndex: number, a: number): HdPath {
...
+The overarching goal for migrations is to avoid the use of the Cosmos Hub path for chains
+other than the Cosmos Hub. This means that the Cosmos Hub does not need to migrate to the
+new Cosmos purpose as long as there is no good reason to do so. We'll cover this as a
+special case below. Chains that decide to register their own SLIP44 coin type and do not
+want to reuse Cosmos client tooling (in particular the Cosmos Ledger app) are also
+discussed separately.
+
+Which derivation path is used cannot easily be detected by the chain or remote clients.
+Only clients that do the derivation know the path. So most of the following migration work
+has to happen in the client or is coordination effort between different clients. The
```
Comment (willclarktech) [inline README.md:322]:
```suggestion
has to happen in the client or requires a coordination effort between different clients. The
```

File: README.md [hunk 4]
```diff
@@ -309,7 +310,46 @@ export function makeSimpleHdPath(chainIndex: number, a: number): HdPath {
...
+following rough guide is supposed to provide an overview, but might not fully apply to
+every ecosystem:
+
+1. Get community support for adopting the Cosmos purpose described above.
+2. Register a chain index.
+3. Get clarity if the simple HD path is what you want to use. The answer is probably yes.
```
Comment (willclarktech) [inline README.md:328]:
```suggestion
3. Get clarity over whether the simple HD path is what you want to use. The answer is probably yes.
```

File: README.md [hunk 5]
```diff
@@ -309,7 +310,46 @@ export function makeSimpleHdPath(chainIndex: number, a: number): HdPath {
...
+   to the user encouraging a change.
+5. If you develop a new client support the new schema only. Consider an import feature
+   that supports legacy paths for users migrating from other clients. Consider account
+   discovery using both path patterns.
+
+### Special case: Cosmos Hub
+
+The BIP-44 compliant path `m/44'/118'/x'/0/y` belongs to coin type 118 (ATOM) and the
+Cosmos Hub can use this path as long as it wants to. The only slight drawback is the 5
+component structure originating from Bitcoin, but no real word problem with that is known
```
Comment (willclarktech) [inline README.md:341]:
```suggestion
component structure originating from Bitcoin, but no real world problem with that is known
```

**Review by willclarktech — COMMENTED** [review]
Very nice migration notes.

File: README.md (hunk 2 above)
Comment (webmaster128) [reply README.md:37]: Yeah, right :/ I guess I need to fork that tool.

**Pull request merged**